dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/constant/DefaultConstantService.java

```diff
@@ -54,12 +60,33 @@
 
     private IdentifiableObjectStore<Constant> constantStore;
 
+    private final CacheProvider cacheProvider;
+
+    private Cache<List> ALL_CONSTANTS_CACHE;
```

Comment:
I think this should use `Cache<List<Constant>>`, name should not be that of a java constant as it is a instance field.

Reply:
I tried exactly that at first, but I couldn't get the cache to return a List<Constant> because I could not cast to a specific type of list. (This is OBE.)

Reply:
You need to go via raw types, try a cast like this: `Cache<List<Constant>> ALL_CONSTANTS_CACHE = (Cache) ...`. 
This will generate a raw types and unchecked warning but you can suppress them and it spares you from casting at the usage site.

Reply:
Done, thanks!